lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2866,8 +2869,45 @@ instance LiftHandler (ErrInvalidDerivationIndex 'Soft level) where
                 , "between ", pretty minIx, " and ", pretty maxIx, " without a suffix."
                 ]
 
+instance LiftHandler ErrSelectionCriteria where
+    handler = \case
+        ErrSelectionCriteriaOutputTokenBundleSizeExceedsLimit e ->
+            handler e
+        ErrSelectionCriteriaOutputTokenQuantityExceedsMaxBound e ->
+            handler e
+
+instance LiftHandler ErrTokenBundleSizeExceedsLimit where
+    handler e = apiError err403 TokenBundleSizeExceedsLimit $ mconcat
+        [ "One of the outputs you've specified contains too many assets. "
+        , "Try splitting these assets across two or more outputs. "
+        , "Destination address: "
+        , pretty (view #address e)
+        , ". Asset count: "
+        , pretty (view #assetCount e)
+        , "."
+        ]
+
+instance LiftHandler ErrTokenQuantityExceedsMaxBound where
+    handler e = apiError err403 TokenQuantityExceedsMaxBound $ mconcat
+        [ "One of the asset quantities you've specified is greater than the "
+        , "maximum quantity allowed in a single transaction output. Try "
+        , "splitting this quantity across two or more outputs. "
```

Comment:
This is because the cap for a single output is `maxBound :: Word64`, but quantities in general are `Natural`?

...yes.

Reply:
That's correct. But the cap only applies to single outputs.

It's quite possible to get more than `maxBound :: Word64` for a given asset if we're summing across a whole UTxO set to find a wallet's balance.